File: 백준/Bronze/1259. 팰린드롬수/팰린드롬수.js
const fs = require("fs");
const filePath =
  process.platform === "linux" ? "/dev/stdin" : "../../input.txt";
const inputs = fs.readFileSync(filePath, "utf-8").trim().split("\n");

/**
 * 투 포인터로 left, right로 검사 시작
 * left > right가 될때까지 수행
 */
const checkPalindrome = (number) => {
  const arr = String(number).split("");
  const length = arr.length;

  let isPalindrome = true;
  let [left, right] = [0, length - 1];
  while (left <= right) {
    // console.log("checking left : ", arr[left], "right : ", arr[right]);

    if (arr[left] !== arr[right]) {
      isPalindrome = false;
      break;
    }
    left += 1;
    right -= 1;
  }

  return isPalindrome;
};

for (const input of inputs) {
  if (Number(input) === 0) {
    break;
  }
  if (checkPalindrome(input) === true) {
    console.log("yes");
  } else {
    console.log("no");
  }
}

File: 백준/Bronze/1259. 팰린드롬수/팰린드롬수.py

import copy
while(True):
  n = int(input())
  if(n == 0):
    break
  
  numbers = []
  while(n > 0):
    numbers.insert(0,n % 10)
    n //= 10
  
  length = len(numbers)
  start,end  = 0, length - 1
  
  isPalindrome = True
  while(start <= end):
    # 정답이 아닌 경우
    if(numbers[start] != numbers[end]):
      isPalindrome = False
      break
    # 이어서 작업
    start += 1
    end -= 1
  if(isPalindrome):
    print("yes")
  else:
    print("no")
      
    
      